cclient/iota_client_extended_api.c

```diff
@@ -249,3 +249,63 @@ retcode_t iota_client_get_account_data(iota_client_service_t const* const serv,
   get_balances_res_free(&balances_res);
   return ret_code;
 }
+
+retcode_t iota_client_find_transaction_objects(
+    iota_client_service_t const* const serv,
+    find_transactions_req_t const* const tx_queries,
+    transaction_array_t out_tx_objs) {
+  retcode_t ret_code = RC_OK;
+  find_transactions_res_t* find_tx_res = find_transactions_res_new();
+
+  log_debug(CCLIENT_EXTENDED_LOGGER_ID, "[%s:%d]\n", __func__, __LINE__);
+  if (!find_tx_res) {
+    ret_code = RC_CCLIENT_NULL_PTR;
+    log_error(CCLIENT_EXTENDED_LOGGER_ID,
+              "%s create request object failed: %s\n", __func__,
+              error_2_string(ret_code));
+    goto done;
+  }
+
+  ret_code = iota_client_find_transactions(serv, tx_queries, find_tx_res);
+  if ((ret_code == RC_OK) && find_tx_res) {
+    ret_code = iota_client_get_transaction_objects(
+        serv, (get_trytes_req_t*)find_tx_res, out_tx_objs);
+  }
+
+done:
+  if (find_tx_res) {
+    find_transactions_res_free(&find_tx_res);
+  }
+  return ret_code;
+}
+
+retcode_t iota_client_get_transaction_objects(
+    iota_client_service_t const* const serv, get_trytes_req_t* const tx_hashes,
+    transaction_array_t out_tx_objs) {
+  retcode_t ret_code = RC_OK;
+  get_trytes_res_t* out_trytes = get_trytes_res_new();
+  hash8019_queue_entry_t* q_iter = NULL;
+  iota_transaction_t tx = NULL;
+
+  ret_code = iota_client_get_trytes(serv, tx_hashes, out_trytes);
+  if (ret_code == RC_OK) {
+    CDL_FOREACH(out_trytes->trytes, q_iter) {
+      tx = transaction_deserialize(q_iter->hash);
```

Comment:
Maybe use `transaction_serialize_on_flex_trits` on a preallocated `flex_trit_t [FLEX_TRIT_SIZE_8019]` to avoid allocation ?

Reply:
you mean `transaction_deserialize_from_trits` right? yes, for avoiding dynamic allocation, I can use `struct _iota_transaction tx` instead of `transaction_deserialize`.

Reply:
oops yes, that was what I meant, still waking up here :)

Reply:
never mind, thanks for your suggestion.